src/Data/Managers/GetInstanceWithRelations/QueryManager.ts

```diff
@@ -83,33 +83,42 @@ export function GetInstanceWithRelationsQueryManager(
 
       if (Either.isRight(instanceWithReferences)) {
         if (instanceWithReferences.value.data.referenced_by !== null) {
-          await Promise.all(
+          const resolvedNestedInstances = await Promise.all(
```

Comment:
There are a few downsides to use promise.all, but I'm not sure it's something we need to care about. 

1. It doesn't timeout, so if one of the operations hangs, it won't abort on it's own. 
2. It's all or nothing, if one element fails, the entire promise will fail. It makes it hard to have individual error-handling.
3. It isn't sequential, but executes all the promises concurrently.